File: blobstore/sample code/python/sample_minio.py
import boto.s3.connection
from boto.s3.key import Key


class MINIO:
    client = None

    def __init__(self, endpoint, access_key, secret_key):
        self.client = boto.connect_s3(
            aws_access_key_id=access_key,
            aws_secret_access_key=secret_key,
            host=endpoint,
            is_secure=True,
            calling_format=boto.s3.connection.OrdinaryCallingFormat()
        )

    # -------list buckets---------------
    def list_bucket(self):
        list = []
        buckets = self.client.get_all_buckets()
        for bucket in buckets:
            list.append(bucket.name)
        return list

    # -------create bucket---------------
    def create_bucket(self, bucket_name):
        self.client.create_bucket(bucket_name)
        print("crete bucket:", bucket_name)

    # -------delete bucket---------------
    def delete_bucket(self, bucket_name):
        self.client.delete_bucket(bucket_name)
        print("delete bucket:", bucket_name)

    # -------list objects---------------
    def list_object(self, bucket_name):
        list = []
        bucketclient = self.client.get_bucket(bucket_name)
        keys = bucketclient.get_all_keys()
        for key in keys:
            list.append(key.name)
        return list

    # -------upload object---------------
    def upload_object(self, bucket_name, object_name, str):
        bucketclient = self.client.get_bucket(bucket_name)
        key = Key(bucketclient)
        key.name = object_name
        key.set_contents_from_string(str)
        print("upload object:", object_name)

    # -------download object---------------
    def download_object(self, bucket_name, object_name):
        bucketclient = self.client.get_bucket(bucket_name)
        key = Key(bucketclient)
        key.name = object_name
        s = key.get_contents_as_string()
        return s

    # -------delete object---------------
    def delete_object(self, bucket_name, object_name):
        bucketclient = self.client.get_bucket(bucket_name)
        key = Key(bucketclient)
        key.name = object_name
        key.delete()
        print("delete object:", object_name)

File: blobstore/sample code/golang-new/golang/minio/sample_minio.go
package minio

import (
	"context"
	"fmt"
	"github.com/minio/minio-go/v7"
	"github.com/minio/minio-go/v7/pkg/credentials"
	"io"
	"log"
	"simple_code/golang/properties"
)

type Bucket struct {
	client *minio.Client
}

func NewMinioClient(param *properties.Credentials) (b Bucket, err error) {
	var credential *credentials.Credentials
	if param.ForceV4 {
		credential = credentials.NewStaticV4(param.AccessKey, param.Secretkey, "")
	} else {
		credential = credentials.NewStaticV2(param.AccessKey, param.Secretkey, "")
	}

	b.client, err = minio.New(param.Endpoint, &minio.Options{
		Creds: credential,
		//openstack needs to connect with v2
		Secure: param.SSL,
	})
	if err != nil {
		log.Println(err)
	}
	return b, err
}

func (b *Bucket) ListBucket() (buckets []string, err error) {
	results, err := b.client.ListBuckets(context.Background())
	if err != nil {
		//HandleError(err)
		log.Println(err)
	}
	for _, result := range results {
		buckets = append(buckets, result.Name)
	}
	return buckets, err
}

func (b *Bucket) CreateBucket(bucketName string) error {
	err := b.client.MakeBucket(context.Background(), bucketName, minio.MakeBucketOptions{})
	if err != nil {
		//HandleError(err)
	}
	return err
}

func (b *Bucket) DeleteBucket(bucketName string) error {
	err := b.client.RemoveBucket(context.Background(), bucketName)
	if err != nil {
		//HandleError(err)
	}
	return err
}

func (b *Bucket) ListObject(bucketName string) (objects []string) {
	var results <-chan minio.ObjectInfo
	results = b.client.ListObjects(context.Background(), bucketName, minio.ListObjectsOptions{})
	for result := range results {
		objects = append(objects, result.Key)
	}
	return objects
}

func (b *Bucket) UploadObject(bucketName string, objectName string, reader io.Reader) error {
	rsp, err := b.client.PutObject(context.Background(), bucketName, objectName, reader, -1, minio.PutObjectOptions{})
	if err != nil {
		//HandleError(err)
	}
	fmt.Println(rsp, "--------------------------")
	return err
}

func (b *Bucket) DownloadObject(bucketName string, objectName string) (object *minio.Object, err error) {
	object, err = b.client.GetObject(context.Background(), bucketName, objectName, minio.GetObjectOptions{})
	if err != nil {
		//HandleError(err)
	}
	return object, err
}

func (b *Bucket) DeleteObject(bucketName string, objectName string) error {
	err := b.client.RemoveObject(context.Background(), bucketName, objectName, minio.RemoveObjectOptions{})
	if err != nil {
		//HandleError(err)
	}
	return err
}
